### PR TITLE
[ENH] Normalise/TimeSeriesScaler reconciliation

### LINKED ISSUE #2075: [ENH] StandardScalar and Normalise Collection transformers
### Describe the feature or idea you want to propose

as @MatthewMiddlehurst pointed out at the time, (I didnt notice), there is duplication of functionality in these two collection transformers. We should resolve it/ Also related to #1210 

### Describe your proposed solution

benchmark both and pick one to keep. 

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs

Fixes #2075 

Result of dev discussion at module meeting on 1/11/24. Removes the TimeSeriesScaler, replaced with three classes in the file _rescaler

Normalizer - maps all series onto N(0,1)
Centerer- shifts all series to zero mean
MinMaxScaler - Maps all series onto [0,1]

allows unequal length and multivariate (independent transform for each channel

- [x] tests
- [x] API
- [x] notebook
- [x] Refactor testing uses of TimeSeriesScaler to Normalizer

not doing here, but I think @dguijo may have wanted it

make invertible

### CODE REVIEW

**Review by hadifawaz1999 — COMMENTED** [review]
i think would be nice to add visual examples of series normalized before and after, either unit tests datas or a non normalized ucr one

**Comment by TonyBagnall** [comment]
> i think would be nice to add visual examples of series normalized before and after, either unit tests datas or a non normalized ucr one

agreed, after #2279

File: aeon/transformations/collection/_rescale.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        self.max = max
+        super().__init__()
+
+    def _transform(self, X, y=None) -> np.ndarray:
+        """
+        Transform method to apply the MinMaxScaler normalisation.
+
+        Parameters
+        ----------
+        X: np.ndarray or list
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:125]: Better to say

X : np.ndarray or list
            Collection to transform. Either a list of 2D arrays with shape
            ``(n_channels, n_timepoints_i)`` or a single 3D array of shape
            ``(n_cases, n_channels, n_timepoints)``.

File: aeon/transformations/collection/_rescale.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                    max_val - min_val == 0, 1, max_val - min_val
+                )  # Prevent division by zero
+                x_scaled = (x - min_val) / range_val
+                x_scaled = self.min + x_scaled * (self.max - self.min)
+                Xt.append(x_scaled)
+            return Xt
+
+
+class Centerer(BaseCollectionTransformer):
+    """Standardarise transformer for collections.
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:166]: centerer transformer

File: aeon/transformations/collection/_rescale.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def _transform(self, X, y=None) -> np.ndarray:
+        """
+        Transform method to standardize series.
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:194]: center series

File: aeon/transformations/collection/_rescale.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            Collection to transform. Either a list of 2D arrays with shape
+            ``(n_channels, n_timepoints_i)`` or a single 3D array of shape
+            ``(n_cases, n_channels, n_timepoints)``.
+        y : None
+            Ignored.
+
+        Returns
+        -------
+        X_transformed : np.ndarray or list
+            The normalized data.
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:208]: centered data ?

File: aeon/transformations/collection/_rescale.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        """
+        if isinstance(X, np.ndarray):
+            # Case 1: X is a single 3D array
+            mean_val = np.mean(X, axis=-1, keepdims=True)
+            std_val = np.std(X, axis=-1, keepdims=True)
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:213]: no need of std division

File: aeon/transformations/collection/_rescale.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+            std_val = np.where(std_val == 0, 1, std_val)  # Prevent division by zero
+            X_standardized = (X - mean_val) / std_val
+            return X_standardized
+
+        else:
+            # Case 2: X is a list of 2D arrays
+            Xt = []
+            for x in X:
+                mean_val = np.mean(x, axis=-1, keepdims=True)
+                std_val = np.std(x, axis=-1, keepdims=True)
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:223]: no need of std division

File: aeon/transformations/collection/tests/test_rescaler.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    X_transformed = normaliser._transform(X)
+
+    mean = np.mean(X_transformed, axis=-1)
+    std = np.std(X_transformed, axis=-1)
+
+    assert np.allclose(mean, 0)
+    assert np.allclose(std, 1)
+
+
+def test_standardize():
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/tests/test_rescaler.py:27]: test_centerer

File: aeon/transformations/collection/_rescale.py [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def _transform(self, X, y=None) -> np.ndarray:
+        """
+        Transform method to apply the seleted normalisation.
```
Comment (hadifawaz1999) [inline aeon/transformations/collection/_rescale.py:39]: to apply the normaliztion, no need for selected

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
just a few minor things

File: aeon/transformations/collection/_rescale.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/transformations/collection/_rescale.py:125]: I dont actually know why we put these doc strings in protected methods

**Review by hadifawaz1999 — DISMISSED** [review]
LGTM  but seems a conflict exists

**Review by hadifawaz1999 — DISMISSED** [review]

**Comment by MatthewMiddlehurst** [comment]
`Normalizer` or `Normaliser`? We have other PRs changing to UK english.

**Comment by TonyBagnall** [comment]
> `Normalizer` or `Normaliser`? We have other PRs changing to UK english.

which other PRs? I went Normaliser then realised the rest of the toolkit is US English

**Pull request merged**